web/src/hotkey.js

```diff
@@ -733,15 +733,29 @@ export function process_hotkey(e, hotkey) {
         return false;
     }
 
-    if (hotkey.message_view_only && popover_menus.is_gear_menu_popover_displayed()) {
-        // Inside the gear menu, we don't process most hotkeys; the
-        // exception is that the gear_menu hotkey should toggle the
-        // menu closed again.
+    // We don't need to process arrow keys in navbar menus for spectators
+    // since they only have gear menu present.
+    if (
+        popover_menus.is_personal_menu_popover_displayed() &&
+        event_name === "left_arrow" &&
+        !page_params.is_spectator
+    ) {
+        // Open gear menu popover on left arrow.
+        $("#personal-menu").trigger("click");
+        gear_menu.toggle();
+        return true;
+    }
```

Comment:
It might be nicer to write this with an outer `if (popover_menus.is_personal_menu_popover_displayed())` and have the rest be a second `if` statement inside, and `return false` at the end of the outer `if` statement, matching how the gear menu popover logic works, if we decide it's correct to have the `return false` there.

If nothing else, it'd make this more nicely a "block for this popover" for readability.